### PR TITLE
Smp restore ut cases

### PR DESCRIPTION
**Pull request by alfred2g**
SMP restore UT cases

Description
-----------
Restore UT test cases after separating them

Test Steps
-----------
Overall coverage rate:
  lines......: 97.2% (2874 of 2958 lines)
  functions..: 100.0% (216 of 216 functions)
  branches...: 92.3% (1705 of 1847 branches)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by chinglee-iot — COMMENTED** [review]
Please consider the following suggestions
* Rebase the repo to restore removed vTaskList test cases
* Update the test brief comment
* Update the test function name with prefix "test_coverage_"

File: FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c [hunk 1]
```diff
@@ -179,3 +179,984 @@ int suiteTearDown( int numFailures )
...
+    /* Entering prvYieldCore */
+    vFakePortCheckIfInISR_ExpectAndReturn( pdTRUE );
+    vFakePortGetCoreID_ExpectAndReturn( 2 );
+    vFakePortGetCoreID_ExpectAndReturn( 2 );
+    vFakePortYieldCore_Expect( 1 );
+    /* Leaving prvYieldCore */
+    vFakePortExitCriticalSection_Expect();
+
+    /* API call */
+    vTaskPreemptionEnable( xTaskHandle );
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c:223]: we could, but it is better to call the public API directly if possible
i can change it if needed

File: FreeRTOS/Test/CMock/smp/config_assert/config_assert_utest.c [hunk 2]
```diff
@@ -178,3 +178,450 @@ void port_release_task_lock_cb( int num_calls )
...
+ *
+ * configASSERT( ( pxTCB->xTaskRunState == xCoreID ) || ( pxTCB->xTaskRunState == taskTASK_YIELDING ) );
+ *
+ * @endcode
+ *
+ * configNMBER_OF_CORES > 1
+ */
+void test_prvSelectHighestPriorityTask_assert_coreid_ne_runstate( void )
+{
+    TCB_t unblockedTCB[ configNUMBER_OF_CORES ] = { 0 };
```
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/config_assert/config_assert_utest.c:381]: It looks like only unblockedTCB[ 0 ] is used in this test.

File: FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c [hunk 3]
```diff
@@ -179,3 +179,978 @@ int suiteTearDown( int numFailures )
...
+        else
+        {
+            TEST_FAIL();
+        }
+    }
+    /* Test Verifications */
+    /* the verification of the test is above in the expectations */
+}
+
+void port_assert_if_isr_cb( int num_callbacks )
```
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c:727]: This function is used in the test case only. Suggest we use static function here and move it to helper function section.

File: FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c [hunk 4]
```diff
@@ -179,3 +179,978 @@ int suiteTearDown( int numFailures )
...
+}
+
+/**
+ * @brief This test ensures that if the notify is zero and the ticks to wait are
+ *        greater than zero, the task is yielded
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *
+ * vTaskYieldWithinAPI();
```
Comment (chinglee-iot) [inline FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c:983]: It is suggested that to include the branch condition which the test wants to verify in the code snippet.
```suggestion
 * if( xTicksToWait > ( TickType_t ) 0 )
 * {
 *         ...
 *         vTaskYieldWithinAPI();
 *         ...
 * }
 * else
 * {
 *     mtCOVERAGE_TEST_MARKER();
 * }
```

**Review by chinglee-iot — DISMISSED** [review]
The implementation looks good to me.

Some suggestions for this PR:
1. Test function name is not aligned ( test_coverage_xxx )
2. More code snippet and description to indicate which branch is take or not. Some of the code snippet shows only branch. Ready may not know if the branch is taken or not.

Other observations:
1. Calling public API to test private API indirectly. I don't think it is a problem for unit test.
2. Comment format is not aligned with other test cases. I think it is fine to me as long as there is a brief, code snippet and clear description.

Approve this PR first since these are not implementation problem and can be fixed later.

**Review by chinglee-iot — APPROVED** [review]
Approve for conflict

**Pull request merged**